lib/routes.js

```diff
@@ -36,10 +36,21 @@ const create = async function (server, { api, basedir, cors, vhost, handlers, ex
 
     for (const [path, operations] of Object.entries(api.paths)) {
         const pathnames = Utils.unsuffix(path, '/').split('/').slice(1).join(SEPARATOR);
+        const hapiHandler = operations['x-hapi-handler'];
 
         for (const [method, operation] of Object.entries(operations)) {
-            const pathsearch = `${pathnames}${SEPARATOR}${method}`;
-            const handler = Hoek.reach(handlers, pathsearch, { separator: SEPARATOR });
+            const operationId = operation.operationId;
+            const hapiHandlerMethod = `${hapiHandler}${SEPARATOR}${method}`;
+            const hapiHandlerOperationId = `${hapiHandler}${SEPARATOR}${operationId}`;
+            const pathsearchMethod = `${pathnames}${SEPARATOR}${method}`;
+            const pathSearchOperationId = `${pathnames}${SEPARATOR}${operationId}`;
+            // Give precedence to operationId over method and give precedence to hapiHandler over path.
+            const handler =
+                Hoek.reach(handlers, hapiHandlerOperationId, { separator: SEPARATOR }) ||
```

Comment:
I guess we are searching for `xxx/Undefined; most of the time and do a disk access